nipoppy/workflows/extractor.py

```diff
@@ -91,6 +91,22 @@ def dpath_pipeline(self) -> Path:
             pipeline_version=self.proc_pipeline_info.VERSION,
         )
 
+    @cached_property
+    def dpath_pipeline_output(self) -> Path:
+        """Return the path to the pipeline's output directory."""
+        return self.layout.get_dpath_pipeline_output(
+            pipeline_name=self.proc_pipeline_info.NAME,
+            pipeline_version=self.proc_pipeline_info.VERSION,
+        )
+
+    @cached_property
+    def dpath_pipeline_idp(self) -> Path:
+        """Return the path to the pipeline's IDP directory."""
+        return self.layout.get_dpath_pipeline_idp(
+            pipeline_name=self.proc_pipeline_info.NAME,
+            pipeline_version=self.proc_pipeline_info.VERSION,
+        )
+
```

Comment:
These do not depend on `dpath_pipeline` so they each have to be overriden to use `proc_pipeline_info`, which is annoying so I made an issue to refactor this